src/format/packed/mod.rs

```diff
@@ -51,18 +42,72 @@ pub enum Variant {
 #[cfg(debug_assertions)]
 impl<T: Element> ::format::Validate for Packed<T> {
     fn validate(&self) {
-        assert_eq!(self.values.len(), storage!(self.size));
+        assert_eq!(self.values.len(), triangular_elems(self.size));
     }
 }
 
 size!(Packed, size, size);
 
+/// Computes the offset into the array for the upper triangular representation
+/// Copied from http://www.netlib.org/lapack/lug/node123.html
```

Comment:
Since the code is otherwise a magic formula, I thought this would make it easier to understand a year or 2 from now--having the source of the layout algorithm referenced will make it possible for future users to debug this code. Do you still want the citation to be removed?